src/test/java/org/apache/commons/compress/compressors/DetectCompressorTest.java

```diff
@@ -164,7 +183,70 @@ public void testDetect() throws Exception {
     }
 
     @Test
-    public void testDetection() throws Exception {
+    public void testDetectNullOrEmptyCompressorNames() throws Exception {
+        try (CompressorInputStream bzip2 = createStreamFor("bla.txt.bz2")) {
+            assertThrows(IllegalArgumentException.class, () -> {
+                CompressorStreamFactory.detect(bzip2, (Set<String>) null);
+            });
+        }
+
+        try (CompressorInputStream gzip = createStreamFor("bla.tgz")) {
+            assertThrows(IllegalArgumentException.class, () -> {
+                CompressorStreamFactory.detect(gzip, new HashSet<>());
+            });
+        }
+    }
+
+    @Test
+    public void testDetectLimitedByName() throws Exception {
+        assertEquals(CompressorStreamFactory.BZIP2, detect("bla.txt.bz2",
+                Collections.singleton(CompressorStreamFactory.BZIP2)));
+        assertEquals(CompressorStreamFactory.GZIP, detect("bla.tgz",
+                Collections.singleton(CompressorStreamFactory.GZIP)));
+        assertEquals(CompressorStreamFactory.PACK200, detect("bla.pack",
+                Collections.singleton(CompressorStreamFactory.PACK200)));
+        assertEquals(CompressorStreamFactory.XZ, detect("bla.tar.xz",
+                Collections.singleton(CompressorStreamFactory.XZ)));
+        assertEquals(CompressorStreamFactory.DEFLATE, detect("bla.tar.deflatez",
+                Collections.singleton(CompressorStreamFactory.DEFLATE)));
+        assertEquals(CompressorStreamFactory.LZ4_FRAMED, detect("bla.tar.lz4",
+                Collections.singleton(CompressorStreamFactory.LZ4_FRAMED)));
+        assertEquals(CompressorStreamFactory.LZMA, detect("bla.tar.lzma",
+                Collections.singleton(CompressorStreamFactory.LZMA)));
+        assertEquals(CompressorStreamFactory.SNAPPY_FRAMED, detect("bla.tar.sz",
+                Collections.singleton(CompressorStreamFactory.SNAPPY_FRAMED)));
+        assertEquals(CompressorStreamFactory.Z, detect("bla.tar.Z",
+                Collections.singleton(CompressorStreamFactory.Z)));
+        assertEquals(CompressorStreamFactory.ZSTANDARD, detect("bla.tar.zst",
+                Collections.singleton(CompressorStreamFactory.ZSTANDARD)));
+    }
+
+    @Test
+    public void testDetectLimitedByNameNotFound() throws Exception {
+        assertThrows(CompressorException.class, () ->
```

Comment:
Reduce duplication. How many `Collections.singleton(CompressorStreamFactory.DEFLATE)` do we really need?